docs/source/traits_user_manual/notification.rst

```diff
@@ -719,3 +719,62 @@ steps other than 1, **index** holds the _slice_ that was changed.
 The TraitDictEvent has an additional **changed** attribute which holds the
 keys that were modified and the _old_ values that those keys held.  The new
 values can be queried from directly from the trait value, if needed).
+
+
+.. _on-trait-change-dos-n-donts:
+
+
+Dos and Don’ts
+--------------
+
+Do not assume handlers are called in a specific order
```

Comment:
Nit: I am going to use "Don't" instead of "Do not", just to be consistent with the title.